modules/localgov_review_date/src/Entity/ReviewDate.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+namespace Drupal\localgov_review_date\Entity;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\scheduled_transitions\Entity\ScheduledTransitionInterface;
+
+/**
+ * Defines the ReviewDate entity.
+ *
+ * @ingroup localgov_workflows
+ *
+ * @ContentEntityType(
+ *   id = "review_date",
+ *   label = @Translation("Review date"),
+ *   base_table = "review_date",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "langcode" = "langcode",
+ *     "uuid" = "uuid",
+ *     "uid" = "author",
+ *   },
+ *   handlers = {
+ *     "views_data" = "Drupal\views\EntityViewsData",
+ *   },
+ *   admin_permission = "administer localgov_review_date",
+ * )
+ */
+class ReviewDate extends ContentEntityBase implements ReviewDateInterface {
+
+  /**
+   * Workflow state that transition content to on the next review date.
+   */
+  public const REVIEW_STATE = 'review';
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function newReviewDate(EntityInterface $entity, string $langcode, ScheduledTransitionInterface $transition, ?AccountInterface $author = NULL, bool $active = TRUE): ReviewDate {
+    if (is_null($author)) {
+      $author = \Drupal::currentUser();
+    }
+
+    $review_date = static::create();
+    $review_date
+      ->setEntity($entity)
+      ->setLangauge($langcode)
+      ->setScheduledTransition($transition)
+      ->setAuthor($author)
+      ->setActive($active)
+      ->setCreatedTime(time());
+
+    return $review_date;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getActiveReviewDate(EntityInterface $entity, string $langcode): ?ReviewDate {
+
+    if (!$entity->id()) {
+      return NULL;
+    }
+
+    $review_date_storage = \Drupal::entityTypeManager()->getStorage('review_date');
```

Comment:
Being picky, but this and later, EntityBase already has the entity type manager.